File: TypeScript/ProjectEuler3.ts
class PrimeList {
    private list: number[] = [];

    constructor(to:number) {
        this.list = [];
        this.findFactor(to);

        this.list = this.list.filter(x => this.isPrime(x));

    }

    private findFactor(num: number) {
        for (var i = 2; i < Math.sqrt(num); i++) {
            if (num % i === 0) {
                this.list.push(i);
            }
        }
    }

    private isPrime(num:number) {
        for (var i = 2; i * i < num; i++) {
            if (num % i === 0) {
                return false;
            }
        }
        return true;
    }

    public getPrimeList() {
        return this.list;
    }
}

var num = 600851475143;
var factorArray = new PrimeList(num).getPrimeList();
console.log(factorArray[factorArray.length - 1]);
while (true) {
    
}

File: Java/ProjectEuler3.java
import java.lang.reflect.Array;
import java.util.*;
import java.util.stream.IntStream;


class Problem3 {
    public static List<Long> findFactor(long number){
        List<Long> intList = new ArrayList<>();
        for (long i = 2; i < (long)Math.sqrt(number); i++) {
            if (number % i == 0){
                System.out.println(i);
                intList.add(i);
            }
        }
        return intList;
    }

    public static boolean isPrime(long number){
        return number == 1 ? false :IntStream.range(2, (int) Math.sqrt(number) + 1).asLongStream().filter(x -> number % x == 0).toArray().length == 0;
    }

}

public class Main {

    public static void main(String[] Args){
        long num = 600851475143L;
        long a = Problem3.findFactor(num).stream().filter(x -> num % x == 0).filter(Problem3::isPrime).max((o1, o2) -> o1 > o2 ? 1 : o2 > o1?-1 :0).get();
        System.out.println(a);

    }

}
